js/about/history.js

```diff
@@ -0,0 +1,128 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Note that these are webpack requires, not CommonJS node requiring requires
+const React = require('react')
+const Immutable = require('immutable')
+const Sticky = require('react-stickynode')
+const ImmutableComponent = require('../components/immutableComponent')
+const messages = require('../constants/messages')
+const aboutActions = require('./aboutActions')
+
+const ipc = window.chrome.ipc
+
+// Stylesheets
+require('../../less/about/itemList.less')
+require('../../less/about/history.less')
+require('../../node_modules/font-awesome/css/font-awesome.css')
+
+class HistoryItem extends ImmutableComponent {
+  navigate () {
+    aboutActions.newFrame({
+      location: this.props.history.get('location'),
+      partitionNumber: this.props.history.get('partitionNumber')
+    })
+  }
+  render () {
+    // Figure out the partition info display
+    let partitionNumberInfo
+    if (this.props.history.get('partitionNumber')) {
+      let l10nArgs = {
+        partitionNumber: this.props.history.get('partitionNumber')
+      }
+      partitionNumberInfo =
+        <span>&nbsp;(<span data-l10n-id='partitionNumber' data-l10n-args={JSON.stringify(l10nArgs)} />)</span>
+    }
+
+    var className = 'listItem'
+
+    // If the history item is in the selected folder, show
+    // it as selected
+    if (this.props.inSelectedFolder) {
+      className += ' selected'
+    }
+
+    return <div role='listitem'
+      className={className}
+      onContextMenu={aboutActions.contextMenu.bind(this, this.props.history.toJS(), 'history')}
+      data-context-menu-disable
+      onDoubleClick={this.navigate.bind(this)}>
+    {
+      this.props.history.get('customTitle') || this.props.history.get('title')
+      ? <span className='aboutListItem' title={this.props.history.get('location')}>
+        <span className='aboutItemDate'>{new Date(this.props.history.get('lastAccessedTime')).toLocaleDateString()}</span>
+        <span className='aboutItemTitle'>{this.props.history.get('customTitle') || this.props.history.get('title')}</span>
```

Comment:
There is a chance the title is blank, pls test and format correctly for that case.